Evaluate lisp expressions within the original buffer's context

apella opened the pull request:
This variable is checked in both `tshell-buffer` and
`tshell-out-buffer`. As a result, this variable cannot be a
buffer-local variable. Otherwise it will always be set to the initial
value when evaluating it in `tshell-elisp-eval`.

apella commented:
Alternatively we could move the evaluation out of the `with-current-buffer` block?

TatriX commented:
> Alternatively we could move the evaluation out of the `with-current-buffer` block?

It should be possible to have multiple shells at the same time, so yes, this seems like a better solution.

Commit:
Evaluate expressions within the original buffer

As `*` and `tshell--current-prompt` are local variables, expressions
using them must be evaluated inside the context of the original buffer
and not within the context of `tshell-out-buffer`.
